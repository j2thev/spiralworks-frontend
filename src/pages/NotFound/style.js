import styled from 'styled-components';
import { Container } from 'reactstrap';

export const NotFoundContainer = styled(Container)`
  margin-top: 50px
`;
