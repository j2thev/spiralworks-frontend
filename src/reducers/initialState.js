export default {
  user: {
    id: '',
    email: '',
    phoneNo: '',
    firstName: '',
    lastName: '',
    isAuthenticated: false
  }
}
