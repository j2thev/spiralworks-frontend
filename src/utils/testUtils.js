import React from 'react';
import { Provider } from 'react-redux';

export function TestProvider({ children, store }) {
  return <Provider store={store}>{children}</Provider>;
}
