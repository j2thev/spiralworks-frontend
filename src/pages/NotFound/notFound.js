import React from 'react';
import {
  Row,
  Col
} from 'reactstrap';
import * as R from './style';

function NotFound() {
  return (
    <R.NotFoundContainer>
      <Row>
        <Col sm="12" md={{ size: 6, offset: 3 }}>
          <h5>Oops!</h5>
          Page not found.
        </Col>
      </Row>
    </R.NotFoundContainer>
  );
}

export default NotFound;

