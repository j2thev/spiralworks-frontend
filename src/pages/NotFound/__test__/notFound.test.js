import React from 'react';
import configuredMockStore from 'redux-mock-store';
import toJSON from 'enzyme-to-json';
import { mount } from 'enzyme';
import { TestProvider } from '../../../utils/testUtils';
import initialState from '../../../reducers/initialState';
import '../../../tools/setupEnzyme';
import NotFound from '../../NotFound';

describe('NotFound Page', () => {
  it('should render correctly', () => {
    const store = configuredMockStore()(initialState);
    const container = mount(
      <TestProvider store={store}>
        <NotFound />
      </TestProvider>
    );

    expect(toJSON(container)).toMatchSnapshot();
  });
});
