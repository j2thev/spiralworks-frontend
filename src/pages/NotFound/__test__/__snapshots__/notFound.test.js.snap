// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page should render correctly 1`] = `
<TestProvider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <NotFound>
      <Styled(Container)>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "gOVJvL",
                "rules": Array [
                  "
  margin-top: 50px
",
                ],
              },
              "displayName": "Styled(Container)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <Container
            className="sc-bdVaJa gOVJvL"
            tag="div"
          >
            <div
              className="sc-bdVaJa gOVJvL container"
            >
              <Row
                tag="div"
              >
                <div
                  className="row"
                >
                  <Col
                    md={
                      Object {
                        "offset": 2,
                        "size": 8,
                      }
                    }
                    sm="12"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="col-sm-12 col-md-8 offset-md-2"
                    >
                      <h3>
                        Oops!
                      </h3>
                      Page not found.
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </StyledComponent>
      </Styled(Container)>
    </NotFound>
  </Provider>
</TestProvider>
`;
